InstantPrintTool.py

```diff
@@ -28,9 +37,11 @@ def __init__(self, iface, populateCompositionFz=None):
         self.pressPos = None
         self.populateCompositionFz = populateCompositionFz
 
-        self.dialog = QDialog(self.iface.mainWindow())
+        self.dialog = InstantPrintDialog()
+        self.dialog.setWindowFlags(Qt.WindowStaysOnTopHint)
```

Comment:
Actually it's better to just pass the parent (`self.iface.mainWindow()`) to the `InstantPrintDialog` constructor (as was done with the `QDialog` before), that way the window manager ensures the dialog is above the qgis main window, but not generally on top of *all* windows.

Reply:
You added the parent argument in InstantPrintDialog which is fine, but you also need to actually pass it here. And remove the WindowStaysOnTopHint flag.